src/pubsub/ua_pubsub_keystorage.c

```diff
@@ -563,6 +564,12 @@ storeFetchedKeys(UA_Client *client, void *userdata, UA_UInt32 requestId,
             goto cleanup;
     }
 
+    if(ks->callBackId != 0) {
```

Comment:
Add a short comment.
It is surprising that the callback is removed from within `storeFetchedKeys`.
When and where will it be re-added?

We could also pull this out rather into the method calling `storeFetchedKeys`.

Reply:
> It is surprising that the callback is removed from within `storeFetchedKeys`.
> When and where will it be re-added?

The previous callback is removed from the queue and new callback is added with a new keylifetime. I added this information in the comments now.